src/sim_explorer/cli/display_results.py

```diff
@@ -0,0 +1,87 @@
+from typing import List
+
+from rich.console import Console
+from rich.panel import Panel
+
+from sim_explorer.models import AssertionResult
+
+console = Console()
+
+
+def reconstruct_assertion_name(result: AssertionResult) -> str:
+    """
+    Reconstruct the assertion name from the key and expression.
+
+    :param result: Assertion result.
+    :return: Reconstructed assertion name.
+    """
+    time = result.time if result.time is not None else ""
+    return f"{result.key}@{result.temporal.name}{time}({result.expression})"
+
+
+def log_assertion_results(results: dict[str, List[AssertionResult]]):
```

Comment:
replace `List` (from typing module) with `list` (Python builtin)
```suggestion
def log_assertion_results(results: dict[str, list[AssertionResult]]):
```
same with the other occurences of `List` in this module